core/dbt/graph/selector_methods.py

```diff
@@ -719,7 +719,9 @@ def check_modified_contract(
     ) -> Callable[[Optional[SelectorTarget], SelectorTarget], bool]:
         # get a function that compares two selector target based on compare method provided
         def check_modified_contract(old: Optional[SelectorTarget], new: SelectorTarget) -> bool:
-            if hasattr(new, compare_method):
+            if new is None and hasattr(old, compare_method + "_deleted"):
```

Comment:
can't say I'm in love with this it was the best way forward I could think of that: 
1. reuses the `state:modified.contract` user-facing method ([configured here](https://github.com/dbt-labs/dbt-core/pull/10221/files#diff-9aa4de7075574ef36d838665148e5858294fae6fd66c0eed1fcf2033a34ee84eR752)) to detect both changes that affect deleted + non-deleted changes. 
2. keep the diff method localized to the node type instead of a more global search logic in the StateSelection method which felt like overloading what it did + deviating from the current pattern too much.

open to other suggestions!